chroma_core/services/rpc.py

```diff
@@ -160,9 +161,21 @@ def translate_error(err):
                 django.db.connection.close()
 
         with self._response_conn_pool[_amqp_connection()].acquire(block=True) as connection:
+            def errback(exc, _):
+                log.info('RabbitMQ rpc got a temporary error. May retry. Error: %r', exc, exc_info=1)
+
+            retry_policy = {
+                'max_retries': 10,
+                'errback': errback
+            }
+
+            connection.ensure_connection(**retry_policy) 
+
             with Producer(connection) as producer:
-                maybe_declare(_amqp_exchange(), producer.channel)
-                producer.publish(result, serializer="json", routing_key=self.body['response_routing_key'], delivery_mode = 1, immedate = True, mandatory = True)
+
+                maybe_declare(_amqp_exchange(), producer.channel, True, **retry_policy)
+                producer.publish(result, serializer="json", routing_key=self.body['response_routing_key'],
+                                 delivery_mode=TRANSIENT_DELIVERY_MODE, retry=True, retry_policy=retry_policy, immedate=True, mandatory=True)
```

Comment:
The formatting looks off here, or is github just rendering it weird?

Reply:
maybe, I used the autoformatter in vscode to reformat it.